### ISSUE TITLE
[QUESTION] How to set up nestjs-session with session-file-store

### ISSUE THREAD

**Issue by FiniteLooper** [iamolegga/nestjs-session#1186]
[x] I've read [the docs](https://github.com/iamolegga/nestjs-session/blob/master/README.md)

[x] I've read [the docs of express-session](https://github.com/expressjs/session/blob/master/README.md)

[x] I couldn't find the same [question](https://github.com/iamolegga/nestjs-session/issues?q=is%3Aissue+label%3Aquestion)

**Question**
I am re-writing a plain Express app into a more structured NestJS app and I am having trouble dealing with sessions.  Here is my setup.

**main.ts**
```ts
import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';

async function bootstrap(): Promise<void> {
  const app = await NestFactory.create(AppModule);

  const port = process.env.PORT;
  await app.listen(port);
}

bootstrap().catch((err: unknown) => {
  console.error(err);
});
```
**app.module.ts**
```ts
import { Module } from '@nestjs/common';
import * as session from 'express-session';
import { SessionModule } from 'nestjs-session';
import * as sessionFileStore from 'session-file-store';

import { AppController } from './app.controller';
import { AppLoggerModule } from './app-logger/app-logger.module';
import { AuthModule } from './auth/auth.module';

const sessionTimeToLiveInSeconds = 60 * 60 * 12; //43200 seconds = 12 days
const cookieMaxAgeInMilliseconds = 1000 * 60 * 60 * 24 * 7; //604800000 ms = 1 week
const FileStore = sessionFileStore(session);

@Module({
  imports: [
    SessionModule.forRoot({
      session: {
        secret: process.env.secret,
        resave: true,
        store: new FileStore({ path: './sessions', ttl: sessionTimeToLiveInSeconds }),
        saveUninitialized: false,
        cookie: {
          secure: false,
          httpOnly: false,
          sameSite: false,
          maxAge: cookieMaxAgeInMilliseconds,
        },
      },
    }),
    AuthModule,
    AppLoggerModule,
  ],
  controllers: [AppController],
  providers: [],
})
export class AppModule {}
```

And when I start the NestJS server it's all good, but I make a simple GET request for something that just returns a `"Hello World"` string which works but I get the following error in my console 5 times

```
[session-file-store] will retry, error on last attempt: Error: ENOENT: no such file or directory, open 'C:\Users\<my-repo>\sessions\jm22rlgjMBLrxR0H3SUayhAlPEXI6oZY.json'
```

And of course no session JSON files are created either. Please help, I'm a little new at this but I feel like I'm doing everything correctly here and my system just can't write files to that folder for some reason. The `sessions` folder does exist, and it has the correct read/write permission on it.

**Please mention other relevant information such as Node.js version and Operating System.**
```
NestJS Version : 10.3.8
OS Version     : Windows 10.0.19045
NodeJS Version : v20.10.0
NPM Version    : 10.2.3

express-session: 1.18.0
nestjs-session : 3.0.1   
```

**Comment by micalevisk** [comment]
what if this error is related with this one from that lib: https://github.com/valery-barysok/session-file-store/issues/106

**Comment by FiniteLooper** [comment]
Well it does seem like that's a match, but the Express app I am migrating from also uses `express-session` and `session-file-store` at the exact same versions I am using in this Nest app. It works fine over there. To me it seems like this should just work

**Comment by FiniteLooper** [comment]
Well, upon reading it a second time I see the error message is the same, but that person says the file **was** created. The files are **not** created in my above example

**Comment by iamolegga** [comment]
@FiniteLooper could you please create two repos with minimal reproducible examples? One for express + session-file-store and the other one for nestjs + session-file-store

**Comment by FiniteLooper** [comment]
So in my daily development I have switch back to our main code branch which still uses the Express server and I can getting the exact same error there!  I feel like this actually might be unrelated to this project now... but I'm not sure what is going on here either.

I was hoping it was some auto-update from `session-file-store` that was breaking things, but I'm on `1.5.0` which was released 3 years ago... so no updates have happened recently.

I was able to change `saveUninitialized` to `true` and this actually fixes my issue. I still get errors in my console, but then it creates the session file and then all requests after that seem to work. I'm really not sure I understand this... but I think I'm good for now.

For anyone that might run into this in the future, this thread was helpful: https://github.com/valery-barysok/session-file-store/issues/41
